Fix failing render tests and use {} instead of Set()

nigelmegitt opened the pull request:
Matches the pull request submitted to the main repo, sandflow/imscJS#212, which catches some cases we hadn't noticed, like there being a region specified on body, or no body present at all, which both would have caused exceptions. Also removes use of `Set()` which has a name clash with a `Set()` object defined in the library, and is an es6 feature.

nigelmegitt committed:
Fix failing render tests and use {} instead of Set()

Matches the pull request submitted to the main repo, which catches some cases we hadn't noticed, like there being a region specified on body, or no body present at all. Also removes use of `Set()` which has a name clash with a `Set()` object defined in the library, and is an es6 feature.

nigelmegitt committed:
Address review comments

Matches 9c3dc7b81b2a10e00b628039fe354131f722b4b6 and rebuilds dist files.

Don't bother with backgroundImage, and define local variables as vars.

JibberJim commented on src/main/js/isd.js:
```diff
@@ -93,27 +116,32 @@
         /* rewritten TTML will always have a default - this covers it. because the region is defaulted to "" */
-        if (activeRegions.size === 0 && tt.head.layout.regions.hasOwnProperty("")) {
-            activeRegions.add("");
+        if (Object.getOwnPropertyNames(activeRegions).length === 1 && activeRegions[""] !== undefined) {
```
As discussed in slack, I'm not sure that this is necessary/useful, and I didn't find it obvious what it was doing, the default region creation has already checked there aren't any other regions, would default the default activeRegion to true in all cases save this check and avoid the confusion / slowness of this check - when there's only one activeRegion the optimisation isn't necessary of which regions are active in any case.

nigelmegitt committed:
Remove unnecessary `getOwnPropertyNames()` call

@JibberJim noted that calling `Object.getOwnPropertyNames()` is costly, and on inspection of the code I realised it is unnecessary. Removing.

nigelmegitt committed:
Rebuild dist files

JibberJim reviewed (approved):
All good.

nigelmegitt committed:
rebuild dist
